contracts/operational/Registry.sol

```diff
@@ -213,33 +164,32 @@ contract Registry is Initializable, SignerOwnable, RegistryGateway, RegistryWork
     //  - Only network can execute this function.
     function perform(
         uint256 workflowId,
+        uint256 workflowExecutionId,
         uint256 gasAmount,
         bytes calldata data,
         address target
-    ) external onlySigner onlyExistingWorkflow(workflowId) {
-        uint256 gasUsed = gasleft();
-
+    ) external override onlySigner {
         // Get a workflow by ID
         Workflow memory workflow = getWorkflow(workflowId);
-        require(workflow.id > 0, "Registry: workflow not found");
 
         // Make sure the workflow is not paused
         require(workflow.status == WorkflowStatus.ACTIVE, "Registry: workflow must be active");
 
-        // Get current balance of workflow owner
-        uint256 currentBalance = getBalance(workflow.owner);
-
-        // Make sure workflow owner has enough funds
-        require(currentBalance > 0, "Registry: not enough funds on balance");
+        require(
+            billingManager.getWorkflowExecutionOwner(workflowExecutionId) == workflow.owner &&
```

Comment:
Also, we have to check if the given workflow ID == workflow ID of the given execution